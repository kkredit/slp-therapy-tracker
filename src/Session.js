import React from 'react';
import './Session.css';

class Session extends React.Component {
  render() {
    return (
      <h1>
        Session page
      </h1>
    );
  }
};

export default Session;
