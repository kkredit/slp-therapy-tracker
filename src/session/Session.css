
#setupCard {
  margin-top: 50px;
  margin-bottom: 10px;
}

#setupCard .setup-hdr {
  margin-top: 10px;
}

.setupButton {
  margin: 15px;
}

.sideNavButton {
  margin-top: 250px;
  height: 70px;
  width: 70px;
}

.sideNavButtonText {
  margin-top: 10px;
}

.dataColHdr {
  text-align: center;
  font-size: 16pt;
}

.nameLocation {
  font-size: 10pt;
}

#dataContainer {
  outline: solid LightSlateGray 3px;
  padding: 10px;
  margin: 10px;
}

.studentName, .goalNum {
  font-size: 10pt;
  font-style: italic;
}

.studentName {
  margin-top: 35px;
  text-align: right;
}

.carouselDirections {
  color: black;
}

.goalsCarousel .carousel-item {
  -webkit-transition: 0.2s;
  -moz-transition: 0.2s;
  -o-transition: 0.2s;
  transition: 0.2s;
}
