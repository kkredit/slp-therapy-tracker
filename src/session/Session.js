import React from 'react'
import './Session.css'
import NotificationAlert from '../helpers/NotificationAlert.js'
import SessionSetup from './SessionSetup.js'
import DataEntry from './DataEntry.js'
import { ATTEMPT_SUCC, ATTEMPT_CUED, ATTEMPT_FAIL } from './Student.js'

class Session extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      db: props.db,
      alertActive: false,
      alertVariant: '',
      alertText: '',
      provider: null,
      location: null,
      inSetup: true,
      students: []
    };
    this.startCallback = this.startCallback.bind(this);
    this.endCallback = this.endCallback.bind(this);
    this.submitSession = this.submitSession.bind(this);
    this.setAlert = this.setAlert.bind(this);
    this.setAlertError = this.setAlertError.bind(this);
    this.clearAlert = this.clearAlert.bind(this);
  }

  static getDerivedStateFromProps(props) {
    return {
      db: props.db
    };
  }

  render() {
    return (
      <React.Fragment>
        <h1> Record a Session </h1>
        <NotificationAlert active={this.state.alertActive}
                           variant={this.state.alertVariant}
                           text={this.state.alertText} />
        <SessionSetup active={this.state.inSetup}
                      startCallback={this.startCallback}
                      providers={this.state.db.providers}
                      locations={this.state.db.locations} />
        <DataEntry active={!this.state.inSetup}
                   students={this.state.students}
                   provider={this.state.provider}
                   location={this.state.location}
                   endCallback={this.endCallback} />
      </React.Fragment>
    );
  }

  startCallback(startStudents, startProvider, startLocation) {
    this.setState({
      students: startStudents,
      provider: startProvider,
      location: startLocation,
      inSetup: false
    });
  }

  endCallback(endProvider, endLocation, endStudents) {
    this.setState({
      provider: endProvider,
      location: endLocation,
      students: endStudents,
      inSetup: true
    });
    console.log("session ended; students = ");
    console.log(endStudents);
    console.log("student 0 goal 0 attempts:");
    const attempts = endStudents[0].goals[0].attempts;
    const succ = attempts.filter(function(a){return a === ATTEMPT_SUCC}).length;
    const cued = attempts.filter(function(a){return a === ATTEMPT_CUED}).length;
    const fail = attempts.filter(function(a){return a === ATTEMPT_FAIL}).length;
    console.log("success: " + succ + ", cued: " + cued + ", fail: " + fail);
    console.log("provider ID = " + endProvider.id + ", location ID = " + endLocation.id);

    this.submitSession(endProvider, endLocation, endStudents);
  }

  submitSession(endProvider, endLocation, endStudents) {
    // const newSession = {
    //   time: new Date(),
    //   provider_id: endProvider.id,
    //   location_id: endLocation.id,
    //   students: endStudents
    // };
    this.state.db.postSession(endLocation, endProvider, endStudents,
                              Function.prototype, Function.prototype);

    // axios
    //   .post(`${API_BASE}/sessions.json`, newSession)
    //   .then(res => {
    //           res.data.key = res.data.id;
    //           console.log("Session submitted!");
    //         })
    //   .catch(err => {
    //            console.log(err);
    //            this.setAlertError(err, "Could not submit the session.");
    //          });
  }

  setAlert(variant, text) {
    this.setState({
      alertActive: true,
      alertVariant: '' + variant,
      alertText: text
    });
  }

  setAlertError(err, text) {
    this.setAlert('danger',
                  <React.Fragment>
                    <i className="fa fa-exclamation-triangle" aria-hidden="true"/><br />
                    {err.message}: {err.response ? err.response.statusText : "No response"}<br />
                    {text}
                  </React.Fragment>
                  );
  }

  clearAlert() {
    this.setState({
      alertActive: false,
      alertVariant: '',
      alertText: ''
    });
  }
};

export default Session;
