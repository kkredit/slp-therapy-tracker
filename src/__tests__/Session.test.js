import React from 'react';
import ReactDOM from 'react-dom';
import Session from '../session/Session.js';
import GetTestDb from '../__test_helpers__/SetupDb.js'
import { create } from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <Session db={GetTestDb()} />
    </BrowserRouter>
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Snapshots',()=>{
  var tree;
  var instance;

  beforeEach(() => {
    let db = GetTestDb();

    tree = create(
      <BrowserRouter>
        <Session db={db} />
      </BrowserRouter>
    )
    instance = tree.root.findByType(Session)._fiber.stateNode;

    instance.setState({
      alertActive: false,
      alertVariant: '',
      alertText: '',
      provider: db.providers[0],
      location: db.locations[0],
      inSetup: true,
      students: []
    });
  });

  afterEach(() => {
    tree = null;
    instance = null;
  });

  test('Initial setup', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  })

  test('Blank data entry when inSetup is false', () => {
    instance.setState({inSetup: false});
    expect(tree.toJSON()).toMatchSnapshot();
  })
})
