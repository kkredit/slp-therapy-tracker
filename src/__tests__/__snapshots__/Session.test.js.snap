// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots Blank data entry when inSetup is false 1`] = `
Array [
  <h1>
     Record a Session 
  </h1>,
  <div
    className="container"
    id="dataContainer"
  >
    <div
      className="row"
    >
      <div
        className="col-1"
      />
      <div
        className="col-2"
      >
        <h3
          className="nameLocation"
        >
           
          sally
           
          foreman
           
        </h3>
      </div>
      <div
        className="col-2"
      >
        <h3
          className="nameLocation"
        >
           
          testroom1
           
        </h3>
      </div>
      <div
        className="col-4"
      />
      <div
        className="col-1"
      >
        <button
          className="endButton btn btn-outline-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-trash"
          />
           Discard
        </button>
      </div>
      <div
        className="col-1"
      >
        <button
          className="endButton btn btn-outline-success"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-save"
          />
           Submit
        </button>
      </div>
      <div
        className="col-1"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col-1"
      >
        <h3
          className="dataColHdr"
        >
           Students 
        </h3>
      </div>
      <div
        className="col-1"
      />
      <div
        className="col-9"
      >
        <h3
          className="dataColHdr"
        >
           Goals 
        </h3>
      </div>
      <div
        className="col-1"
      />
    </div>
  </div>,
]
`;

exports[`Snapshots Initial setup 1`] = `
Array [
  <h1>
     Record a Session 
  </h1>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      />
      <div
        className="col-2"
      >
        <a
          className="sideNavButton btn btn-light btn-lg"
          href="/history"
          onClick={[Function]}
          type="button"
        >
          <i
            aria-hidden="true"
            className="fa fa-2x fa-fw fa-angle-double-left"
          />
        </a>
      </div>
      <div
        className="col-4"
      >
        <div
          className="card bg-secondary border-primary"
          id="setupCard"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
               Setup a new session 
            </div>
            <p
              className="card-text"
            >
              Select how many students will be in this session and how many goals each will have.
            </p>
            <div
              className="mb-2 setup-hdr card-subtitle h6"
            >
               Provider 
            </div>
            <select
              className="form-control"
              name="provider"
              onChange={[Function]}
            >
              <option>
                sally
                 
                foreman
              </option>
            </select>
            <div
              className="mb-2 setup-hdr card-subtitle h6"
            >
               Location 
            </div>
            <select
              className="form-control"
              name="location"
              onChange={[Function]}
            >
              <option>
                testroom1
              </option>
            </select>
            <div
              className="mb-2 setup-hdr card-subtitle h6"
            >
               Number of students 
            </div>
            <button
              className="setupButton btn btn-primary btn-sm"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-minus"
              />
            </button>
            1
            <button
              className="setupButton btn btn-primary btn-sm"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-plus"
              />
            </button>
            <div
              className="mb-2 setup-hdr card-subtitle h6"
            >
               Number of goals 
            </div>
            <div>
              Student 
              1
              :
              <button
                className="setupButton btn btn-primary btn-sm"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-minus"
                />
              </button>
              1
              <button
                className="setupButton btn btn-primary btn-sm"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-plus"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-2"
      >
        <button
          className="sideNavButton btn btn-success btn-lg"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <i
            aria-hidden="true"
            className="fa fa-2x fa-fw fa-angle-right"
          />
        </button>
      </div>
      <div
        className="col"
      />
    </div>
  </div>,
]
`;
